accounts-db/src/tiered_storage/index.rs

```diff
@@ -99,13 +99,14 @@ impl IndexBlockFormat {
         index_offset: IndexOffset,
     ) -> TieredStorageResult<AccountOffset> {
         match self {
-            Self::AddressAndOffset => {
+            Self::AddressAndBlockOffsetOnly => {
                 let account_offset = footer.index_block_offset as usize
                     + std::mem::size_of::<Pubkey>() * footer.account_entry_count as usize
-                    + index_offset.0 * std::mem::size_of::<u64>();
-                let (account_block_offset, _) = get_type(mmap, account_offset)?;
+                    + index_offset.0 * std::mem::size_of::<u32>();
+                let (block_offset, _) = get_type::<u32>(mmap, account_offset)?;
```

Comment:
We do need it for now, as the receiver (AccountOffset) is still `usize` and not a `u32`.

This will be fixed together with the PR changing the AccountOffset to `u32`.